app/scripts/components/common/map/style-generators/hooks.ts

```diff
@@ -17,42 +18,59 @@ interface AssetUrlReplacement {
   to: string;
 }
 
-interface UseDatasetTilesOptions {
-  id: string,
-  stacCol: string,
-  tileApiEndpointToUse?: string,
-  stacApiEndpointToUse?: string,
-  date: Date,
-  onStatusChange: CallableFunction,
-  sourceParams?: Record<string, any>,
-  assetUrlReplacements?: AssetUrlReplacement,
-  datasetType?: 'vector' | 'raster' | 'titiler-cmr' | 'zarr' | 'cmr-stac';
+interface DatasetTilesOptions {
+  id: string;
+  stacCol: string;
+  date: Date;
+  onStatusChange: CallableFunction;
+  sourceParams?: Record<string, any>;
 }
 
+interface TilesOptionsWithEndpoints extends DatasetTilesOptions {
+  tileApiEndpointToUse: string;
+  stacApiEndpointToUse: string;
+}
+
+interface CRMSTACOptions extends TilesOptionsWithEndpoints {
+  assetUrlReplacements?: AssetUrlReplacement
+}
+
+interface MosaicOptions extends DatasetTilesOptions {
+  tileApiEndpointToUse: string;
+  datasetType?: DatasetLayerType;
+}
+
+interface StacCollectionOptions extends DatasetTilesOptions {
+  stacApiEndpointToUse: string;
+  datasetType?: DatasetLayerType;
+}
+
```

Comment:
🆒 ! I didn't know about being able to create an interface that is an extension of another